dashboard/src/pages/apps/[id]/settings.tsx

```diff
@@ -401,6 +424,182 @@ export default () => {
     )
   }
 
+  const EnvVarConfigContainer: Component = () => {
+    const [envVars, { refetch: refetchEnvVar }] = createResource(
+      () => app().id,
+      (id) => client.getEnvVars({ id }),
+    )
+
+    const EditEnvVarContainer: Component<{
+      envVar: ApplicationEnvVar
+    }> = (props) => {
+      const [isEditing, setIsEditing] = createSignal(false)
+      let formRef: HTMLFormElement
+      let keyInputRef: HTMLInputElement
+      let valueInputRef: HTMLInputElement
+
+      const handleUpdateEnvVar: JSX.EventHandler<HTMLButtonElement, MouseEvent> = async (e) => {
+        // prevent default form submit (reload page)
+        e.preventDefault()
+
+        // validate form
+        if (!formRef.reportValidity()) {
+          return
+        }
+
+        try {
+          // delete old env var
+          await client.deleteEnvVar({
+            applicationId: app().id,
+            key: props.envVar.key,
+          })
+          // create new env var
+          await client.setEnvVar({
+            applicationId: app().id,
+            key: keyInputRef.value,
+            value: valueInputRef.value,
+          })
```

Comment:
[should] Setという名前の通り削除しなくても更新できたはずです
```suggestion
          await client.setEnvVar({
            applicationId: app().id,
            key: keyInputRef.value,
            value: valueInputRef.value,
          })
```

Reply:
keyを変更する場合にsetのみを使うと古いkeyの物が残ってしまうため、削除後にset、としていました

Reply:
あーなるほどです
その場合はユーザーに削除→追加というステップを踏ませてもいい気がする

Reply:
どうしましょうかね？
keyは変更不可にしてしまっても良いと思ってます

Reply:
大抵のPaaSは環境変数設定画面でKeyも編集できるよな～と思ってこのような実装をしていたのですが、確かにそもそもKeyを変更する機会ってあまりない気がしてきたので変更不可にしてSetのみを使う実装にしてみたいと思います